Add sanitization for custom email settings string

SudharakaP opened the pull request:
Resolve https://github.com/opencollective/opencollective/issues/5776 

![image](https://user-images.githubusercontent.com/12435965/180504655-59d463af-a242-44b3-ac69-dde10a0e611d.png)

https://user-images.githubusercontent.com/12435965/180504782-092a0df0-a00a-41fc-a953-658bbc4431da.mp4

Betree commented on server/lib/collectivelib.ts:
```diff
@@ -148,11 +149,10 @@ export function validateSettings(settings: any): string | boolean {
-  if (settings.customEmailMessage && settings.customEmailMessage.length > 511) {
+  if (settings.customEmailMessage && stripHTML(settings.customEmailMessage).length > 500) {
+    console.log(stripHTML(settings.customEmailMessage));
     return 'Message should be less than 500 characters';
```
Would be great to be more explicit on this one.

```suggestion
    return 'Custom "Thank you" email message should be less than 500 characters';
```

Betree commented on server/lib/search.js:
```diff
@@ -178,7 +178,7 @@ export const searchCollectivesInDB = async (
 
   if (tags?.length) {
-    searchedTags = `{${tags.map(tag => `"${tag}"`).join(',')}}`;
+    searchedTags = `{${tags.map(tag => `"${tag.toLowerCase()}"`).join(',')}}`;
```
I don't get why we're changing this here?

SudharakaP replied:
Sorry about this one, I was trying this out for a different issue and somehow accidentally checked in.

Betree reviewed (commented):
Looks like the right approach but there are a few things to resolve still

SudharakaP commented on server/graphql/v2/mutation/AccountMutations.ts:
```diff
@@ -99,7 +115,8 @@ const accountMutations = {
 
         const settings = account.settings ? cloneDeep(account.settings) : {};
-        set(settings, args.key, args.value);
+        const value = sanitizeSettingsValue(args.key, args.value);
```
Ah, makes sense. I've moved it to the Collective mode. 👍🏽

Betree reviewed (commented):
Tests are failing because `settings` can be null.

I've added an additional test to make sure the feature works.

SudharakaP commented:
> Tests are failing because `settings` can be null.
> 
> I've added an additional test to make sure the feature works.

Thanks for the review and the additional check. I've fixed the problem. 👍🏽
